cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/Range.java

```diff
@@ -0,0 +1,143 @@
+ *  arising from the use of the          accessoire ou fortuit, résultant
+ *  software.  Neither the name          de l'utilisation du logiciel. Ni
+ *  of the National Research             le nom du Conseil National de
+ *  Council of Canada nor the            Recherches du Canada ni les noms
+ *  names of its contributors may        de ses  participants ne peuvent
+ *  be used to endorse or promote        être utilisés pour approuver ou
+ *  products derived from this           promouvoir les produits dérivés
+ *  software without specific prior      de ce logiciel sans autorisation
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fits.slice;
+
+import java.util.Iterator;
+
+
+/**
+ * Represents a Range of integer values.  Can be expanded to an entire array, but only on demand.  This uses concepts
+ * from Python's numpy project to be able to have a start (lowerBound), stop (upperBound), and a striding value (step).
+ */
+public class Range {
```

Comment:
This class name could be more specific, maybe PixelRange?

The 3-arg ctor should call `this(lowerBounds, upperBound)` then just assign `step`.

That single-arg constructor seems like it will inevitably be ambiguous: if there is a use case for (,upper] then there is also a use case for [lower,) ... the correct solution is probably to use the two-arg ctor and change the type to Integer so that null values can be used for open range (and then carefully handle them once you know the actual dimensions). If you can just drop that 1-arg ctor then ignore what's below.

If you do need open-ended range, then it has to be Integer and null allowed. The best practice that we've been using in other code to good effect is to declare thing that can be null as public fields, so if Integer is the correct thing to use here then lowerBound and upperBound would be 
```
public final Integer lowerBound;
public final Integer upperBound;
```
(yeah, public final looks wierd but works as expected).